### PR TITLE
CI: Add spell checking and links checking

### PR DESCRIPTION
**Pull request by rasmusgo**
### Checklist
* [x] I have read and agree to [Contributor Guide](../CONTRIBUTING.md) and the [Code of Conduct](../CODE_OF_CONDUCT.md)

This adds spell checking and broken links checking as github actions.

Spell checking can be enabled locally in vscode with [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) and/or run in batch mode with [cspell](https://cspell.org/).

### CODE REVIEW

**Review by emilk — APPROVED** [review]
omg this is amazing - thank you Rasmus! ❤️

**Comment by emilk** [comment]
Now why isn't the action run on this PR? 🤔

**Comment by rasmusgo** [comment]
Do you need to activate github actions for this repo perhaps?

**Comment by emilk** [comment]
image

This repository is set up exactly like our main `rerun` repo, so I don't wee why this doesn't run…

**Comment by rasmusgo** [comment]
Could it be that there currently are no actions in the main repo (only in my fork) and that makes github think no workflows should be run?

**Comment by emilk** [comment]
Let's see what happens when we merge!

File: .github/workflows/markdown.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: CI (Markdown)
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
```
Comment (emilk) [inline .github/workflows/markdown.yaml:7]: Oh right, it was configured to only run on `main` - I'll add a `pull_requests:` to it too

**Pull request merged**